src/main/java/com/gregtechceu/gtceu/api/machine/trait/RecipeLogic.java

```diff
@@ -477,13 +481,15 @@ public void onRecipeFinish() {
                     lastRecipe.matchTickRecipe(this.machine).isSuccess() &&
                     lastRecipe.checkConditions(this).isSuccess()) {
                 setupRecipe(lastRecipe);
+                if (isActive) consecutiveRecipes++;
             } else {
                 if (suspendAfterFinish) {
                     setStatus(Status.SUSPEND);
                     suspendAfterFinish = false;
                 } else {
                     setStatus(Status.IDLE);
                 }
+                consecutiveRecipes = 0;
```

Comment:
Should also reset if the old recipe isn't the same as the new recipe, no?

Reply:
does it now